shared/test/showcase/echo_test.rb

```diff
@@ -20,11 +20,100 @@
 class EchoTest < ShowcaseTest
   def test_echo
     client = Google::Showcase::V1alpha3::Echo.new(
-      credentials: GRPC::Core::Channel.new(
-        "localhost:7469", nil, :this_channel_is_insecure))
+      credentials: GRPC::Core::Channel.new("localhost:7469", nil, :this_channel_is_insecure)
+    )
 
     response = client.echo content: 'hi there!'
 
     assert_equal 'hi there!', response.content
   end
+
+  def test_chat_closure
+    client = Google::Showcase::V1alpha3::Echo.new(
+      credentials: GRPC::Core::Channel.new("localhost:7469", nil, :this_channel_is_insecure)
+    )
+
+    pull_count = 0
+
+    stream_input = Google::Gax::StreamInput.new
+
+    client.chat stream_input do |response|
+      puts "PULL #{response.content}"
+      pull_count += 1
+
+      sleep rand
+
+      msg, count = response.content.split ":"
+      count = count.to_i
+      count += 1
+      puts "PUSH #{msg}:#{count}"
+      stream_input.push content: "#{msg}:#{count}"
+    end
+
+    Thread.new do
+      10.times do |n|
+        puts "PUSH hello #{n}:1"
+        stream_input.push content: "hello #{n}:1"
+        sleep rand
+      end
+    end
+
+    25.times do
+      break if pull_count >= 20
+
+      sleep 1
+    end
+
+    stream_input.close
```

Comment:
What will happen if methods are called on `stream_input` after the rpc has terminated (i.e. if it errors or aborts)? Would it be useful to have a way to check if the call is active?

Reply:
I don't know how `StreamInput` would be able to determine the health of an RPC call using it's enumerable.

Reply:
Before we talk about how, do you think it's needed? 

The use case I'm thinking of is something like this:

```ruby
until stream_input.closed?
  stream_input.push # next item
end
```

but this could not depend soley on the user `close`ing it without being confusing IMHO.

Reply:
Calling `#close` is no guarantee that the client has pulled all the requests from the input and closed the stream. We can track whether `#close` has been called, and perhaps not allow more requests to be pushed to it, but we cannot determine the state of the stream from the input object.

Reply:
Right, that's really my point though. This interface does not provide a way to monitor the lifecycle of a streaming call. Is this input helper enough or should we also provide a way to do that?

Reply:
I don't know how to track the health of the streams, much less how to design an intuitive API to expose it. AFAIK, as long as the enums are active the stream is active.

Reply:
Well, we could figure that out :), but let's just table this for now. I'd like to see what others think before getting caught up on this (which I'm sort of importing from what's been done elsewhere and may not be a fit here).